core/theme/ThemeManager.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace luya\theme;
+
+use luya\base\PackageConfig;
+use luya\helpers\Json;
+use Yii;
+use luya\Exception;
+use yii\base\InvalidConfigException;
+
+/**
+ * Core theme manager for LUYA.
+ *
+ * This component manage the actual display themes.
+ *
+ * @property Theme $activeTheme
+ *
+ * @author Mateusz Szymański Teamwant <zzixxus@gmail.com>
+ * @author Bennet Klarhölter <boehsermoe@me.com>
+ * @since 1.0.0
+ */
+class ThemeManager extends \yii\base\Component
+{
+    const APP_THEMES_BLANK = '@app/themes/blank';
+    
+    /**
+     * @var Theme
+     */
+    private $_activeTheme;
+    
+    /**
+     * @var ThemeConfig[]
+     */
+    private $_themes = [];
+
+    /**
+     * Setup active theme
+     */
+    public function setup()
+    {
+        if (!$this->activeTheme) {
+            $basePath = $this->getActiveThemeBasePath();
+        } elseif (is_string($this->activeTheme)) {
+            $basePath = $this->activeTheme;
+        } elseif ($this->activeTheme instanceof Theme) {
+            // Active theme already loaded
+            return;
+        }
+    
+        $themeConfig = new ThemeConfig($basePath);
+        $this->activeTheme = new Theme($themeConfig);
+        
+        Yii::$app->view->theme = $this->activeTheme;
+        
+        Yii::setAlias('theme', $this->activeTheme->basePath);
+    }
+
+    /**
+     * Get current theme from SQL
+     * If you have own theme you must inset into db record named active_theme like that:
+     *
+     * INSERT INTO `admin_config` (`name`, `value`, `is_system`, `id`) VALUES ('active_theme', '@app/themes/teamwant', '1', NULL);
+     *
+     * todo: create admin module to manage themes
+     * @return string
+     * @throws \yii\db\Exception
+     */
+    public function getActiveThemeBasePath()
+    {
+        try {
+            $connection = Yii::$app->getDb();
+            $command = $connection->createCommand("SELECT * FROM `admin_config` WHERE `name` = :name limit 1;", [':name' => 'active_theme']);
```

Comment:
LUYA core either does not require Database component or admin module. But i can see there is a todo, so am not sure there will be some redesign on this.